service-base/src/test/java/org/eclipse/hono/service/credentials/AbstractCompleteCredentialsServiceTest.java

```diff
@@ -147,17 +136,15 @@ public void testGetCredentialsFailsForWrongClientContext(final TestContext ctx)
      * @param ctx The vert.x test context.
      */
     @Test
-    public void testRemoveCredentialsByAuthIdAndTypeSucceeds(final TestContext ctx) {
+    public void testRemoveCredentialsByAuthIdAndTypeSucceeds(final VertxTestContext ctx) {
 
         register(getCompleteCredentialsService(), "tenant", "device", "myId", "myType", ctx);
 
-        final Async remove = ctx.async();
-        getCompleteCredentialsService().remove("tenant", "myType", "myId", ctx.asyncAssertSuccess(s -> {
-            assertThat(s.getStatus(), is(HttpURLConnection.HTTP_NO_CONTENT));
+        getCompleteCredentialsService().remove("tenant", "myType", "myId", ctx.succeeding(s -> ctx.verify(() -> {
+            assertEquals(HttpURLConnection.HTTP_NO_CONTENT, s.getStatus());
             assertNotRegistered(getCompleteCredentialsService(), "tenant", "myId", "myType", ctx);
```

Comment:
Does this helps with "nested" calls issues we experienced with Infinispan client?

Reply:
did not tried yet !

Reply:
No, I don't think it would. Just thinking maybe we can tackle this at the same time. You can always use your local builds (SNAPSHOTS) to test against Infinispan. But we can do it, one step at the time.

Reply:
I'll definitely give it a try :)